test/e2e/grpc_test.go

```diff
@@ -211,14 +190,24 @@ func TestGRPCStreamingPing(t *testing.T) {
 
 func TestGRPCUnaryPingFromZero(t *testing.T) {
 	testGRPC(t, func(t *testing.T, names test.ResourceNames, clients *test.Clients, host, domain string) {
-		waitForScaleToZero(t, names, clients)
+		deploymentName := names.Revision + "-deployment"
+
+		if err := WaitForScaleToZero(t, deploymentName, clients); err != nil {
+			t.Fatalf("Could not scale to zero: %v", err)
+		}
+
 		unaryTest(t, names, clients, host, domain)
 	})
 }
 
 func TestGRPCStreamingPingFromZero(t *testing.T) {
 	testGRPC(t, func(t *testing.T, names test.ResourceNames, clients *test.Clients, host, domain string) {
-		waitForScaleToZero(t, names, clients)
+		deploymentName := names.Revision + "-deployment"
```

Comment:
Why not use rnames.Deployment() here and above like the file below?

Reply:
rnames.Deployment works with the full Revision resource. There was a prior version where I used the name of the revision to fish out the Deployment, so I could pass it to rnames.Deployment(), but the decision was made to just use the string directly since we already had the critical bit of information.

@markusthoemmes noted that "We should rework these helper functions to just take strings so we don't need to go fetch entire resources if we already have the critical information at hand."